mlflow/models/utils.py

```diff
@@ -1146,6 +1159,36 @@ def _enforce_object(data: Dict[str, Any], obj: Object, required=True):
     return data
 
 
+def _enforce_map(data: Any, map_type: Map, required=True):
+    if not required and data is None:
+        return None
+
+    if not isinstance(data, dict):
+        raise MlflowException(f"Expected data to be a dict, got {type(data).__name__}")
+
+    if not all(isinstance(k, str) for k in data):
+        raise MlflowException(f"Expected all keys in the map type data are string type.")
+
+    if isinstance(map_type.value_type, DataType):
+        data_enforced = {k: _enforce_datatype(v, map_type.value_type) for k, v in data.items()}
```

Comment:
Could we add a helper function or a mapping of value_type to the function to map `DataType` -> `_enforce_datatype` etc.. This sort of structure is used in this file a lot :)